[#556] Add changelog (#732)
zcash/ZcashLightClientKit: CHANGELOG.md

# Unreleased
- [#556] Change data structures which represent transactions.

    These data types are gone: `Transaction`, `TransactionEntity`, `ConfirmedTransaction`, 
`ConfirmedTransactionEntity`. And these data types were added: `ZcashTransaction.Overview`, 
`ZcashTransaction.Received`, `ZcashTransaction.Sent`. 

>     New data structures are very similar to the 

Reviewer: I think this line got cut off

> old ones. But not everything is the same. The APIs of the `SDKSynchronizer` stayed the same. 

Reviewer: ```suggestion
old ones. Although there many breaking changes. The APIs of the `SDKSynchronizer` remain unchanged in their behavior.
```

> Those are just using new data types. **When adapting this change check which data types are

Reviewer: ```suggestion
They return different data types. **When adopting this change, you should check which data types are
```

> used by methods of the `SDKSychronizer` that you use and you are good to go.**

Reviewer: ```suggestion
used by methods of the `SDKSynchronizer` that are used in your code and change them accordingly.**
```

>     New transaction structures don't have `memo` property. To get memos for the transaction 

Reviewer: ```suggestion
    New transaction structures no longer have a `memo` property. To get memos for the transaction 
```

the `SDKSynchronizer` has now new methods to fetch those:
    - `func getMemos(for transaction: ZcashTransaction.Overview) throws -> [Memo]`, 
    - `func getMemos(for receivedTransaction: ZcashTransaction.Received) throws -> [Memo]`
    - `func getMemos(for sentTransaction: ZcashTransaction.Sent) throws -> [Memo]`

>     

Reviewer: ```suggestion
    This responds to the fact that Zcash transactions can have either none or multiple memos. 
    You can find out how many memos a `ZcashTransaction.Overview` in its corresponding
    property.
```

# Unreleased
- [#556] Change data structures which represent transactions.

    These data types are gone: `Transaction`, `TransactionEntity`, `ConfirmedTransaction`, 
`ConfirmedTransactionEntity`. And these data types were added: `ZcashTransaction.Overview`, 
`ZcashTransaction.Received`, `ZcashTransaction.Sent`. 

>     New data structures are very similar to the  old ones. Although there many breaking changes.

Reviewer: ```suggestion
    New data structures are very similar to the old ones. Although there many breaking changes.
```

The APIs of the `SDKSynchronizer` remain unchanged in their behavior. They return different 
data types. **When adopting this change, you should check which data types are used by methods 

> of the `SDKSynchronizer` that are used in your code and change them accordingly.**

Reviewer: ```suggestion
of the `SDKSynchronizer` in your code and change them accordingly.**
```
